### PR TITLE
Abdulmalek week3

### PR DESCRIPTION
**Pull request by Abdulmalekhmed**
abdulmalek week3 javascript

### CODE REVIEW

File: 1-JavaScript/Week3/homework/ex1-giveCompliment.js [hunk 1]
```diff
@@ -18,19 +18,25 @@ Full description at: https://github.com/HackYourFuture/Homework/tree/main/1-Java
    Use `console.log` each time to display the return value of the
    `giveCompliment` function to the console.
 -----------------------------------------------------------------------------*/
-function giveCompliment(/* TODO parameter(s) go here */) {
-  // TODO complete this function
+function giveCompliment(str) {
+  let compliments =['great', 'awesome','good','hero','marvellous','fantastic','terrific','grand','brilliant','excellent'];
+  let randomCompliment= compliments[Math.floor(Math.random()*compliments.length)]
```
Comment (bnhoogendijk) [inline 1-JavaScript/Week3/homework/ex1-giveCompliment.js:23]: Very well! I like how you used the length of the compliments array to calculate the random index.

File: 1-JavaScript/Week3/homework/ex2-dogYears.js [hunk 2]
```diff
@@ -16,8 +16,10 @@ calculate it!
    ages.
 -----------------------------------------------------------------------------*/
 
-function calculateDogAge(/* TODO parameter(s) go here */) {
+function calculateDogAge(age) {
   // TODO complete this function
+ let dogage= age*7;
```
Comment (bnhoogendijk) [inline 1-JavaScript/Week3/homework/ex2-dogYears.js:21]: Excellent implementation! Minor thingy: how about some camelCase for the variable name 🐫

File: 1-JavaScript/Week3/homework/ex3-tellFortune.js [hunk 3]
```diff
@@ -33,29 +33,32 @@ body, this code is now written once only in a separated function.
...
+function selectRandomly(array) {
+  let x= array[Math.floor(Math.random()*array.length)]
+  return x;
 }
 
-function tellFortune(/* TODO add parameter(s) here */) {
-  // TODO complete this function
+function tellFortune(numKids,partnerNames,locations,jobTitles ){
+
+  return `you will be ${selectRandomly(jobTitles)} in ${selectRandomly(locations)} marraid to ${selectRandomly(partnerNames)} with ${selectRandomly(numKids)} kids`;
```
Comment (bnhoogendijk) [inline 1-JavaScript/Week3/homework/ex3-tellFortune.js:43]: Nice use of template literals 👏

File: 1-JavaScript/Week3/homework/ex4-shoppingCart.js [hunk 4]
```diff
@@ -23,9 +23,25 @@ you have more than 3 items in your shopping cart the first item gets taken out.
...
+    shoppingCart.push(parameter );}
+   if(shoppingCart.length > 3) {
+     
+     shoppingCart.shift();
+    
+   }
+  
+ 
+   const str =`You bought ${shoppingCart.join(',')}!`;
+   return str;
```
Comment (bnhoogendijk) [inline 1-JavaScript/Week3/homework/ex4-shoppingCart.js:37]: minor detail: you could consider returning the string directly in stead of storing it in a variable first. Apart from that, it works like a charm :-)

File: 1-JavaScript/Week3/homework/ex5-shoppingCartPure.js [hunk 5]
```diff
@@ -16,10 +16,17 @@ it pure. Do the following:
 5. Confirm that you function passes the provided unit tests.
 ------------------------------------------------------------------------------*/
 // ! Function under test
-function addToShoppingCart(/* TODO parameter(s) go here */) {
-  // TODO complete this function
+function addToShoppingCart(shoppingCart, grocey) {
+ 
+ const newShoppingCart = [...shoppingCart, grocey];
+ if (newShoppingCart .length >3){
```
Comment (bnhoogendijk) [inline 1-JavaScript/Week3/homework/ex5-shoppingCartPure.js:22]: well done! just for fun: could you think of a way to return directly without an if statement? something like:
```javascript
function addToShoppingCart(shoppingCart, grocery) {
    return [...shoppingCart, grocery] // change this line
}
```

File: 1-JavaScript/Week3/homework/ex6-totalCost.js [hunk 6]
```diff
@@ -21,22 +21,37 @@ instead!
...
+  banana: 2.5,
 };
 
-function calculateTotalPrice(/* TODO parameter(s) go here */) {
-  // TODO replace this comment with your code
+function calculateTotalPrice(cartForParty){
+  let total = 0;
+  for (let val cartForParty){
+    total+= cartForParty[val];
+  }
```
Comment (bnhoogendijk) [inline 1-JavaScript/Week3/homework/ex6-totalCost.js:35]: I don't think this is proper code. Did you try to run it? I think you are trying to loop over the keys/values in the cartForPlay. You could do something like:

```javacript
for (key in object) {
    // do something with key
}
```
or 
```
Object.keys(object).forEach(key => {
    // do something with key
}
```

File: 1-JavaScript/Week3/homework/ex7-mindPrivacy.js [hunk 7]
```diff
@@ -30,10 +30,15 @@ const employeeRecords = [
 ];
 
 // ! Function under test
-function filterPrivateData(/* TODO parameter(s) go here */) {
-  // TODO complete this function
+function filterPrivateData(employeeRecords) {
+  // const { name, occupation} =employeeRecords;
+  let arr[];
```
Comment (bnhoogendijk) [inline 1-JavaScript/Week3/homework/ex7-mindPrivacy.js:35]: There = something missing in line 35 🙂

File: sysinfo.json [hunk 8]
```diff
@@ -0,0 +1,62 @@
+{
```
Comment (bnhoogendijk) [inline sysinfo.json:1]: I think you committed a little too much. No worries for now, but try and make clean commits with only the changes you intended.

**Review by bnhoogendijk — APPROVED** [review]
Great job on this weeks homework! Most of it works and I left you some comments that I felt fit. Please take a look at my comment for ex6 and 7. I will approve the pr because overall you did well 🎉

**Comment by Abdulmalekhmed** [comment]
hello Bash thanks for spending time to review my home work and for the useful feedback i applied them

File: 1-JavaScript/Week3/homework/ex5-shoppingCartPure.js (hunk 5 above)
Comment (Abdulmalekhmed) [reply 1-JavaScript/Week3/homework/ex5-shoppingCartPure.js:22]: I am not sure if i understand your question , heres a solution as far as i understand the question using the "ternary operator "           condition ? do this if ture   :   do this if false;

**Pull request closed without merging**